pkg/autohealing/cloudprovider/openstack/provider.go

```diff
@@ -288,15 +289,40 @@ func (provider CloudProvider) firstTimeRepair(n healthcheck.NodeInfo, serverID s
 			// Uncordon the node
 			if n.IsWorker {
 				nodeName := n.KubeNode.Name
-				newNode := n.KubeNode.DeepCopy()
-				newNode.Spec.Unschedulable = false
-				if _, err := provider.KubeClient.CoreV1().Nodes().Update(context.TODO(), newNode, metav1.UpdateOptions{}); err != nil {
-					log.Errorf("Failed to cordon node %s, error: %v", nodeName, err)
-				} else {
-					log.Infof("Node %s is cordoned", nodeName)
+				// timeout for wait.Poll
+				ctx := context.Background()
+				errServiceUp := wait.PollUntilContextTimeout(ctx, 3*time.Second, provider.Config.RebuildDelayAfterReboot, false,
+					func(ctx context.Context) (bool, error) {
+						rapairedNode, getErr := provider.KubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Comment:
I wonder why `context.TODO()` when we have the `ctx` here? Just a question, I'm honestly not sure which should be used.

Reply:
Yes, here we should use existing context object